donkeycar/pipeline/types.py

```diff
@@ -71,27 +78,68 @@ class TubDataset(object):
     """
 
     def __init__(self, config: Config, tub_paths: List[str],
-                 shuffle: bool = True) -> None:
+                 seq_size: int = None) -> None:
         self.config = config
         self.tub_paths = tub_paths
-        self.shuffle = shuffle
         self.tubs: List[Tub] = [Tub(tub_path, read_only=True)
                                 for tub_path in self.tub_paths]
         self.records: List[TubRecord] = list()
         self.train_filter = getattr(config, 'TRAIN_FILTER', None)
+        self.seq_size = seq_size
+
+    def get_records(self):
+        if not self.records:
+            msg = f'Loading tubs from paths {self.tub_paths} with filter ' \
+                  f'{self.train_filter}' if self.train_filter else ''
+            logger.info(msg)
+            for tub in self.tubs:
+                for underlying in tub:
+                    record = TubRecord(self.config, tub.base_path, underlying)
+                    if not self.train_filter or self.train_filter(record):
+                        self.records.append(record)
+            if self.seq_size:
+                seq = Sequencer(self.seq_size, self.records)
+                self.records = list(seq)
+        return self.records
+
+
+class Sequencer(Iterable[List[TubRecord]]):
```

Comment:
Consider calling this `Collator` because that is what other ML frameworks use.

Reply:
Done.